jetbrains-core/src/software/aws/toolkits/jetbrains/ui/wizard/python/SamInitProjectBuilderPyCharm.kt

```diff
@@ -31,7 +34,14 @@ class SamInitProjectBuilderPyCharm : PythonProjectGenerator<PyNewProjectSettings
 
     override fun getLogo() = AwsIcons.Resources.SERVERLESS_APP
 
-    override fun validate(baseDirPath: String) = settingsPanel.validate()
+    override fun validate(baseDirPath: String): ValidationResult {
+        val currSamPath = settingsPanel.samExecutableField.text
+        if (currSamPath != lastSamPath) {
```

Comment:
Should we be checking the timestamp on the file too? What if they've upgraded and the path is the same but the executable is new?

Reply:
also it looks like we do not have the UI in-view once `beforeProjectGenerated` is called...